rts/motoko-rts/src/idl.rs

```diff
@@ -11,6 +11,11 @@ use core::cmp::min;
 
 use motoko_rts_macros::ic_mem_fn;
 
+extern "C" {
+    // check instruction decoding limit, exported by moc
+    pub fn idl_limit_check(value_count: u32);
```

Comment:
Should we maybe consider using `u64` just in case to avoid/defer overflows (in the compiler backend limit calculation)? Maybe not a concern.

Reply:
Done